x-pack/packages/kbn-random-sampling/src/ui/slider_control/index.tsx

```diff
@@ -70,11 +87,19 @@ export function ControlSlider({
             step={1}
             min={0}
             max={values.length - 1}
-            ticks={values.map((v, i) => ({
-              // Remove the initial 0 from values with decimal digits: 0.001 => .001
-              label: `${v * 100}%`.slice(Number.isInteger(v * 100) ? 0 : 1),
-              value: i,
-            }))}
+            ticks={values.map((v, i) => {
+              let accessibleLabel: string | undefined;
+              if (v === 0.00001 || v === 1) {
+                accessibleLabel = tickLabels[v];
+              }
+
+              return {
+                // Remove the initial 0 from values with decimal digits: 0.001 => .001
+                label: `${v * 100}%`.slice(Number.isInteger(v * 100) ? 0 : 1),
+                value: i,
+                accessibleLabel,
+              };
+            })}
```

Comment:
This breaks the logic of a reusable component mixing some precondition (the concept that the label 1 and 0.0001 are the extents of your values, but this can't be always true considering that we can pass an `values: number[]` ) Now, for example, I can't create an random sampling slider with values that are 10 - 0 because labels like 1 are automatically converted into a logic that doesn't depend on the values array.

I'd like to suggest to change the `values` array to be converted to `values: Array<{label: string, value: number, accessibleLabel?: string}>` and give the consumer the ownership of providing the right labels for the right set of values.

Reply:
🙆🏼‍♀️  122ba08e3468cc610251458fc85d730903a96f1b